cmd/mailer.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/services/mailer"
+
+	"github.com/urfave/cli"
+)
+
+func runSendMail(c *cli.Context) error {
+	if err := argsSet(c, "title"); err != nil {
+		return err
+	}
+
+	if err := initDB(); err != nil {
+		return err
+	}
+
+	users, _, err := models.SearchUsers(&models.SearchUserOptions{
+		Type:        models.UserTypeIndividual,
+		OrderBy:     models.SearchOrderByAlphabetically,
+		ListOptions: models.ListOptions{},
+	})
+	if err != nil {
+		return errors.New("Cann't find users")
+	}
+
+	var emails []string
+	for _, user := range users {
+		emails = append(emails, user.Email)
```

Comment:
Don't we risk exposing all emails? Does this use BCC?

Reply:
I am not sure, but I as understand after going through the code, gitea uses `net/smtp`. And this is a method which responsible for this. (As I understand )

https://golang.org/pkg/net/smtp/#Client.Rcpt

There is a sort of specification I found.

https://blog.mailtrap.io/smtp-commands-and-responses/#RCPT_TO

>The RCPT TO command specifies the recipient. As an argument, RCPT TO includes a destination mailbox (forward-path). In case of multiple recipients, RCPT TO will be used to specify each recipient separately. 

So probably it will send each separately though I was previously thinking that it will do behave in a CC maner

Reply:
There's more I found in RFC 

So a huge list of reciprocates may cause a error.

> If an SMTP server has an implementation limit on the number of RCPT
   commands and this limit is exhausted, it MUST use a response code of
   452 (but the client SHOULD also be prepared for a 552, as noted
   above).  If the server has a configured site-policy limitation on the
   number of RCPT commands, it MAY instead use a 5yz response code.

And according to your question. So the behavior is literally implementation depended.
> 7.2.  "Blind" Copies
    SMTP clients and servers SHOULD NOT copy
   the full set of RCPT command arguments into the header section,
   either as part of trace header fields or as informational or private-
   extension header fields.  Since this rule is often violated in
   practice, and cannot be enforced, sending SMTP systems that are aware
   of "bcc" use MAY find it helpful to send each blind copy as a
   separate message transaction containing only a single RCPT command.

Reply:
59a1bee